File: Java/program50.java
///////////////////////////////////////////////////////
//
//  File name :     program50.java
//  Descreption :   Separate Digit and Do Summation of Digits from Input Number
//  Author :        Vaibhav
//  Date :          20/05/2025
//
///////////////////////////////////////////////////////

import java.util.Scanner;

class CountEven
{
    int iValue;

    int Display()
    {       
        int iDigit = 0;
        int iSum = 0;
        
        while(iValue != 0)
        {   
            iDigit = iValue % 10; 
            iSum = iSum + iDigit;
            iValue = iValue / 10;
        }
        return iSum;
    }
}   
class program50
{
    public static void main(String A[])
    {
        int iNo = 0;
        int iRet = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter Number : ");
        iNo = sobj.nextInt();

        System.out.println("-----------------------------------------------------------------------------------------------");
        System.out.println("Number Received is "+iNo);
        System.out.println("-----------------------------------------------------------------------------------------------");
        

        CountEven cobj = new CountEven();

        cobj.iValue = iNo;

        iRet = cobj.Display();

        System.out.println("Summation of Digits of Number "+iNo+" is "+iRet);
    }
}

File: program50.c
///////////////////////////////////////////////////////
//
//  File name :     program50.c
//  Descreption :   Separate Digit and Do Summation of Digits from Input Number
//  Author :        Vaibhav
//  Date :          
//
///////////////////////////////////////////////////////

#include<stdio.h>

int SumDigits(int iNo)
{
    int iDigit = 0;
    int iSum = 0;

    if(iNo < 0)
    {
        iNo = -iNo;
    }
    while(iNo != 0)
    {   
        iDigit = iNo % 10;
        iNo = iNo / 10;
        iSum = iSum + iDigit;
    }
    return iSum;
}

int main()
{
    int iValue = 0;
    int iRet = 0;
    
    printf("Enter number: \n");
    scanf("%d",&iValue);
    
    printf("-----------------------------------------------------------------------------------------------\n");
    printf("Number Received is %d\n",iValue);
    printf("-----------------------------------------------------------------------------------------------\n");

    iRet = SumDigits(iValue);

    printf("Addition of Digits are : %d\n",iRet);
    
    return 0;
}